graftm/bootstrapper.py

```diff
@@ -86,22 +102,47 @@ def generate_hmm_from_contigs(self, contig_files, output_hmm_file):
             # Now have a fasta file of ORFs.
             # Check to make sure the file is not zero-length
             orfs.flush()
-            if os.stat(orfs.name).st_size == 0:
-                logging.warn("Failed to find any matching ORFs in the bootstrap contigs")
+            
+            if len(seqio.read_fasta_file(orfs.name)) <= 1:# Just to build on this, you need to check if there is > 1 hit
+                                                          # otherwise mafft will fail to align, causing a crash when hmmbuild is 
+                                                          # run on an empty file.
+                logging.warn("Failed to find two or more matching ORFs in the bootstrap contigs")
                 return False
             
-            # Run mafft to align them
             with tempfile.NamedTemporaryFile(prefix="graftm_bootstrap_aln") as aln:
-                cmd = "mafft --auto %s >%s" % (orfs.name, aln.name)
-                logging.info("Aligning bootstrap hits..")
-                logging.debug("Running alignment cmd: %s" % cmd)
-                extern.run(cmd)
-            
-                # Run hmmbuild to create an HMM
-                cmd = "hmmbuild --amino %s %s >/dev/null" % (output_hmm_file, aln.name)
-                logging.info("Building HMM from bootstrap hits..")
-                logging.debug("Running cmd: %s" % cmd)
-                extern.run(cmd)
+                    
+                if search_method == "hmmsearch":
+                    # Run mafft to align them
+                    
+                    cmd = "mafft --auto %s >%s" % (orfs.name, aln.name)
+                    logging.info("Aligning bootstrap hits..")
+                    logging.debug("Running alignment cmd: %s" % cmd)
+                    extern.run(cmd)
+                
+                    # Run hmmbuild to create an HMM
+                    cmd = "hmmbuild --amino %s %s >/dev/null" % (output_database_file, aln.name)
+                    logging.info("Building HMM from bootstrap hits..")
+                    logging.debug("Running cmd: %s" % cmd)
```

Comment:
these debug statements are not needed, since extern will log a debug message anyway